.travis.yml

```diff
@@ -54,10 +60,16 @@ jobs:
       before_install: *use-cmake
       env: *Debug
 
-before_script:
-  - export PATH=$(echo "$PATH" | sed -e 's/:\/usr\/lib\/ccache//')
+# Populate the CMake cache from Travis cache, if any
+# Unfortunately Travis can only cache directories, not single files
+before_script: mv CachedCMakeCache/CMakeCache.txt . || true
```

Comment:
It still represents an improvement, think of the cumulative time saved by people waiting on pulls to build! And you also must consider the reduced logspam, and the fact that the sizes of types as stored in the cache don't change for the majority of the builds that are run, so most of the configure process is indirectly contributing to global warming.

Reply:
Correctness beats performance. `CMakeCache.txt` contains a hell of a lot more than just the size of fundamental types.

Reply:
I'm not convinced. The build from scratch ship has already sailed given the use of ccache, and CMake<i>Cache</i>.txt sounds like a prime candidate for cash given that it doesn't change in the vast majority of builds Travis does.

Reply:
>The build from scratch ship has already sailed given the use of ccache,

Not true, `ccache` preserves correctness since it only uses the cache when the pre-compiled source, compiler and compile flags are identical. It even detects things like the `__TIME__` macro and prevents caching.

>CMakeCache.txt sounds like a prime candidate for cash given that it doesn't change in the vast majority of builds Travis does.

It can change significantly if you edit the `CMakeLists.txt` files or a submodule updates their cmake. The only way to trust CI in these scenarios would be to clear the cache.

Reply:
I'll defer to your judgement, I think I can see that if say, someone updates a CACHE'd variable set() but doesn't supply FORCE the new value won't be seen by CI.